sdk/core/System.Net.ClientModel/src/ModelReaderWriter/ModelReaderWriterFormat.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ComponentModel;
+
+namespace System.Net.ClientModel
+{
+    /// <summary>
+    /// A format used to specify how a model should be read and written.
+    /// </summary>
+    public readonly partial struct ModelReaderWriterFormat : IEquatable<ModelReaderWriterFormat>
+    {
+        internal const string JsonValue = "J";
+        internal const string WireValue = "W";
```

Comment:
Do we know if there is a word that maps better to the BCL lexicon than `Wire`?

There is a nice article that maps out BCL terminology in this space that I spent some time with recently here: https://learn.microsoft.com/en-us/dotnet/fundamentals/networking/http/httpclient  It was very helpful in getting my head around how BCL things about HTTP network operations and what words they use to describe the various aspects of that.

Reply:
Not that I know of do you have a suggestion?

Reply:
Not offhand, but I will think about it.  Mostly just flagging that this could come up in the BCL review.

Reply:
Ok, I have an idea.

This is the one that is "whatever the service expects", is that correct?

If so, what about `ServiceContract`?

For me, "wire" communicates the transmission mechanism, e.g. "I send my message over the wire," and less about the piece that receives it on the other side of the wire.  It feels a bit like a throwback to land-line telephone communications.

`ServiceContract` feels like it communicates better the intent of "what the service expects."  It is the contract with the service.  I believe BCL uses the word service in the same way, so this is at least consistent there, and if they have a better word for "contract," this name implies enough that they could proactively tell us that in a review, without having to ask us to explain what "wire" means first.

I also think it goes better alongside "Json" in the enum values, because a customer looking at those side-by-side might wonder "hey, the service contract using JSON, so this other Json must mean something else," and might be able to logic it out themselves, and understand it completely with a boost from good refdoc comments.  Otherwise, it feels to me that people could get confused between whether to use "Wire" or "Json" when the service contract calls for JSON -- I know I was confused about that at first.

What do you think?